### PR TITLE
Avoid problematic encodings in Zlib specs

### PR DESCRIPTION
**Pull request by mjago**
Avoid problematic encodings in Zlib specs

### CODE REVIEW

File: library/zlib/deflate/deflate_spec.rb [hunk 1]
```diff
@@ -1,22 +1,24 @@
-# -*- encoding: us-ascii -*-
```
Comment (eregon) [inline library/zlib/deflate/deflate_spec.rb]: Isn't the magic encoding comment enough for `parser` to figure out that literal strings are ASCII-8BIT, that is BINARY?

Comment (eregon) [inline library/zlib/deflate/deflate_spec.rb]: Interesting, so the strings automatically become `ASCII-8BIT` on MRI if they are not valid `US-ASCII`. But this doesn't happen if `utf-8` is used or there is no comment.

**Comment by eregon** [comment]
This is good, even though not as readable, it avoids relying on very specific semantics of the effect of the magic encoding comment on string literals. Actually, some of these specs fail without the magic encoding comment.
So :+1: and let's get this merged, thanks!

File: library/zlib/deflate/deflate_spec.rb (hunk 1 above)
Comment (mjago) [inline library/zlib/deflate/deflate_spec.rb]: @eregon parser reads the magic encoding comment and then converts from that encoding to utf-8 just fine. It is later (in the lexer I suppose) where invalid strings are rejected "escape sequences incompatible with UTF-8"

**Comment by mjago** [comment]
👍

**Comment by mjago** [comment]
@eregon Let me run this past you. The parser gem reads the "ascii-8bit" magic encoding string, converts the 'file string' from ascii-8bit to utf-8 (encoding_valid? == true) and then throws away the magic string encoding. However at this point the 'file string' is still just a set of utf-8 characters including the characters [",\,x,8,0,"] - the encoding has changed and the original encoding has been thrown away. Later on, when it comes to parsing, the magic string is `skipped' , and the escaped string is parsed as having an encoding of utf-8. Therefore magic encoding is being ignored when it comes to actual source code strings. This is how it appears to work from looking at the code.

**Comment by mjago** [comment]
@eregon Oh yes, the file /  `file string' is simply:
``
# encoding: ascii-8bit
"\x80"
''

**Pull request merged**